### PR TITLE
Make it easier to contribute

### PR DESCRIPTION
**Pull request by nullpixel**
# Work done:

* Refactored configuration files
* Made control messages configurable
* Moved the auth stuff to a discord group
* Uses a domain from config to generate archive URLs
* Updated README with steps on getting started with contribution 

# Motivation

The old example config lacked a lot of values needed to get an instance of the bot up and running. This pull request makes it somewhat easier to get started.

# Known Gripes

I load the configuration file manually twice in this pull request, which to me is somewhat unacceptable, however, I could not come up with a better way of creating what I wanted from it.

---

It's worth nothing I'd like to work on the web panel, however, there is no contributor chat where I would be able to collaborate with you on this. Rowboat is a super cool bot, so I'd love to contribute somewhat.

### CODE REVIEW

**Comment by guuurgle** [comment]
If you are looking for some sort of contributor chat: https://discordapp.com/invite/rowboat

**Comment by nullpixel1** [comment]
Banned, lol.

File: README.md [hunk 1]
```diff
@@ -8,6 +8,16 @@ Rowboat is a private Discord bot focused on being a highly powerful and configur
 
 Probablly not. Rowboat has enough moving pieces that running a local version is complicated. The main purpose of having the source released is to allow others to understand and audit the functionality. The code is by no means meant to be easy to setup or bootstrap, and I don't plan on supporting folks trying to run locally. That said, feel free to run a local version of Rowboat for your server (but not a public version please).
 
+### Notes to run locally:
+Rowboat depends on a lot of third party tools. The example configuration file outlines all things you will need, but these.
```
Comment (b1naryth1ef) [inline README.md]: nit: newline

File: README.md [hunk 2]
```diff
@@ -8,6 +8,16 @@ Rowboat is a private Discord bot focused on being a highly powerful and configur
...
+
+* Install redis, and start it on the default port
+* Install postgresql, and make two databases, `rowboat`, and `rowboat_stats`. Give the user `rowboat` access to these.
```
Comment (b1naryth1ef) [inline README.md]: Should specify PG 9.6+, also rowboat_stats is going away

Comment (b1naryth1ef) [inline README.md]: I would also just specify the SQL to execute here:

```
CREATE USER rowboat;
CREATE DATABASE rowboat WITH owner rowboat;
```

Also probably worth mentioning the `pg_hba.conf` requirements

File: config.example.yaml [hunk 3]
```diff
@@ -3,9 +3,28 @@ token: ''
 manhole_enable: true
 manhole_bind: 127.0.0.1:7171
 
+DSN: # Get this from here https://docs.sentry.io/quickstart/
```
Comment (b1naryth1ef) [inline config.example.yaml]:
```yaml
DSN: '' #...
```

technically not required afaik

File: config.example.yaml [hunk 4]
```diff
@@ -3,9 +3,28 @@ token: ''
...
+
+SECRET_KEY: # Generate a strong key used for oauth
```
Comment (b1naryth1ef) [inline config.example.yaml]: same here, default to empty string pls

File: config.example.yaml [hunk 5]
```diff
@@ -3,9 +3,28 @@ token: ''
...
+
 state:
   sync_guild_members: false
 
+control_messages:
```
Comment (b1naryth1ef) [inline config.example.yaml]: should be `control_channels`

File: rowboat/models/message.py [hunk 6]
```diff
@@ -197,7 +199,10 @@ def create_from_message_ids(cls, message_ids):
 
     @property
     def url(self):
-        return 'https://rowboat.party/archive/{}.txt'.format(self.archive_id)
+	with open('config.yaml', 'r') as f:
```
Comment (b1naryth1ef) [inline rowboat/models/message.py]: spacing is fucked here, also ideally this should be extracted into a common area, but I can do that post-pr

Comment (nullpixel1) [inline rowboat/models/message.py]: Okay, will fix that

File: config.example.yaml (hunk 4 above)
Comment (nullpixel1) [inline config.example.yaml]: 👍

File: config.example.yaml (hunk 3 above)
Comment (nullpixel1) [inline config.example.yaml]: Ah, crashed for me when the key wasn't there.

File: README.md (hunk 2 above)
Comment (gerd2002) [inline README.md]: The pg_hba.conf requirements are not that hard, and to mention them would mean to copypaste the exact spacing.

Comment (nullpixel1) [inline README.md]: Re-reading this section now, I agree with you. I'll most likely split it up into two sections

Comment (nullpixel1) [inline README.md]: @gerd2002 Not that hard makes no sense? The database needs the `hstore` extension, which I didn't know when I was setting it up

Comment (gerd2002) [inline README.md]: @nullpixel1 that is in the SQL, not in the pg_hba.conf. that file is containing the settings for remote access

Comment (nullpixel1) [inline README.md]: Misread that, lol.
Still worth pointing it out regardless

**Comment by nullpixel1** [comment]
@b1naryth1ef any chance of a re-review?

**Comment by b1naryth1ef** [comment]
I spent some time on the weekend making this process easier with docker/docker-compose. I'll try to merge in some of your readme and config changes later tonight or tomorrow

**Comment by nullpixel** [comment]
sounds good

**Comment by ThaTiemsz** [comment]
Works very well for me. Perhaps adding the emojis to the config would be helpful too.

**Comment by nullpixel** [comment]
closed, will make into new PR if needed

**Pull request closed without merging**